### DISCUSSION TITLE
How to close a Dialog?

### DISCUSSION THREAD

**Discussion by sheikalthaf** [spartan-ng/spartan#194] [Q&A]
I'm currently using the dialog https://www.spartan.ng/components/dialog but i don't know how to close the dialog from ts after successful api call.

**Comment by sheikalthaf** [comment]
@goetzrobin your help will be appreciated here

**Comment by elite-benni** [comment]
With the new Dialog PR, wich will be released soon, you can take the example from the dialog-dynamic-component.
https://github.com/ty-ler/spartan/blob/a515543fd4400f0d0e587ec93c17ca4972a215a3/apps/app/src/app/pages/(components)/components/(dialog)/dialog-dynamic-component.preview.ts

```	
    private readonly _dialogRef = inject<BrnDialogRef<ExampleUser>>(BrnDialogRef);
    .
    .
    this._dialogRef.close(user);
```

for now you can access the dialogComponent via viewChild and close it with this reference. 
```
    @ViewChild('MyDialog') myDialog!: BrnAlertDialogComponent;
    this.myDialog.close()
```

**Comment by goetzrobin** [comment]
You can also use a signal in your TS that you pass into the component. Then you manage the component showing or not showing by setting the value of the signal. Btw the code that enables @elite-benni suggestion was released as part of the latest alpha
